.github/actions/merge-develop-and-set-up-dependencies/action.yml

```diff
@@ -31,6 +35,25 @@ runs:
         key: ${{ runner.os }}-proddeps-${{ hashFiles('/home/runner/work/.pyversion', 'requirements_dev.txt', 'scripts/install_python_dev_dependencies.py') }}
         path: |
           /opt/hostedtoolcache/Python/3.9.20/x64
+
+    # TODO (#22567): Fix pip not working properly when fetched from cache.
+    - name: Check pip installation and fix if needed
```

Comment:
Check pip installation --> Check cached pip installation

One question, what if there was no cache to fetch in the previous step? Would this step then fail?

Reply:
If there is no cache in the previous step, setup-python@v5 will install pip fresh. So, this step will still run, check pip status and as pip won't be corrupted, it will run else block in if-else statement.

Done